src/app/features/users-page/users-page.component.ts

```diff
@@ -1,25 +1,74 @@
-import { Component } from '@angular/core';
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
+import { FilterModel } from '@features/filter-bar/interfaces/filter-model';
 import { IUsers } from '@features/users-page/interfaces/users';
 import { getUsersPending } from '@features/users-page/store/users.actions';
 import { getUsersFromState } from '@features/users-page/store/users.selectors';
 import { Store } from '@ngrx/store';
 import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
 @Component({
+    changeDetection: ChangeDetectionStrategy.OnPush,
     selector: 'crx-users-page',
     styleUrls: ['./users-page.component.scss'],
     templateUrl: './users-page.component.html'
 })
 
-export class UsersPageComponent {
+export class UsersPageComponent implements OnInit {
 
+    @Input() filters: any;
+
+    public filterIsOpen = false;
     public users$: Observable<IUsers[]>;
 
     constructor (private store: Store) {
 
         this.store.dispatch(getUsersPending());
+
+    }
+
+    ngOnInit () {
+
         this.users$ = this.store.select(getUsersFromState);
 
     }
 
+    openFilter () {
+
+        return this.filterIsOpen = !this.filterIsOpen;
+
+    }
+
+    applyFilters (event: FilterModel) {
+
+        if (event.byGender) {
```

Comment:
what if user wants to set female above 18 years old?
you need to allow multiple filters to be applied at same time